src/k8s-extension/azext_k8s_extension/partner_extensions/CostExport.py

```diff
@@ -0,0 +1,248 @@
+    _register_resource_provider(cmd, "Microsoft.CostManagementExports")
+    export_type = 'Usage'
+    # TODO: 'AmortizedCost'
+    args = [
+        "costmanagement", "export", "create",
+        "--name", cluster_name,
+        "--scope", f"/subscriptions/{subscription}/resourceGroups/{mc_resource_group}",
+        "--timeframe", "MonthToDate",
+        "--type", export_type,
+        "--recurrence-period", f"from={datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S')}",
+        "to=2200-01-01T00:00:00",
+        "--recurrence", "Daily",
+        "--schedule-status", "Active",
+        "--storage-account-id", storage_account_id,
+        "--storage-container", "cost",
+        "--query", "'id'",
+        "--subscription", subscription,
+        "-o", "tsv"
+    ]
+    _invoke(args)
+    logger.info("cost export created")
+
+
+def _mc_resource_group(subscription: str, resource_group_name: str, cluster_name: str) -> str:
+    cli = _cli()
+    cli.invoke(
+        ["aks", "show", "--subscription", subscription, "--resource-group", resource_group_name, "-n", cluster_name,
+         "-o", "json"])
+    if cli.result.exit_code != 0:
+        raise Exception("Unable to get cluster resource group")
+    return cli.result.result['nodeResourceGroup']
+
+
+def _ensure_storage_container(storage_account_id: str, storage_container: str):
+    resource = parse_resource_id(storage_account_id)
+    try:
+        _invoke(["storage", "account", "show", "--ids", storage_account_id])
+    except SystemExit as e:
+        if e.code != 3:
+            raise e
+        create_storage_args = ["storage", "account", "create",
+                               "--resource-group", resource['resource_group'],
+                               "--name", resource['name'],
+                               "--subscription", resource['subscription'],
+                               "--allow-blob-public-access", "false"]
+        _invoke(create_storage_args)
+        logger.info("created storage account %s", storage_account_id)
+    cli = _invoke(["storage", "container", "exists", "--name", storage_container, "--account-name", resource["name"],
+                   "--auth-mode", "login"])
+    if cli.result.result['exists']:
+        return
+    _invoke(["storage", "container", "create", "--name", storage_container, "--account-name", resource["name"],
+             "--auth-mode", "login"])
+    logger.info("created new container %s for %s", storage_container, storage_account_id)
+
+
+class ServicePrincipal(TypedDict):
+    appId: str
+    password: str
+    tenant: str
+    displayName: str
+
+
+def _create_service_principal(sp_name: str) -> ServicePrincipal:
+    cli = _invoke(["ad", "sp", "create-for-rbac", "--display-name", sp_name, "--years", "2"])
+    logger.info("created service principal %s", sp_name)
+    return cli.result.result
+
+
+def _cli() -> AzCli:
+    return get_default_cli()
+
+
+def _invoke(args) -> AzCli:
+    cli = _cli()
+    try:
+        cli.invoke(args)
```

Comment:
I wonder whether it's the best way to do so. But couldn't find a better example.